### PR TITLE
FIX: do not update the network queryparam immediately just on subsequent c…

### PR DESCRIPTION
**Pull request by tokdaniel**
## **Summary of Changes**
The network queryparam needs to update when the user switches network, but not immediately upon arriving on the leverage site, since it would cause the queryParams to always take on the user's wallet chainId, instead of prompting them on trading to switch network.

## **Test Data or Screenshots**
The following things should be tested:

`[link]/leverage`
- going here, shoud automatically fill up the `buy`, `sell`, and `network` queryParams, where network should be the user's default network.
`[link]/leverage?buy=ETH2X&sell=ETH&network=42161`
- Opening the link above with a different chainId than what you are connected with
- Opening the link with the same chainId you have connected with
- Opening the link, and trying to do a transaction, after signing, it should prompt to switch on the right network, and changing the network parameter
- Upon switching networks in wallet, or in the app selector, it should override the queryparams if there is a network queryparam set.
`[link]/leverage?buy=ETH2X&sell=ETH&network=`
- opening a link like this, should also fill up the network with the user's network

###### _By submitting this pull request, you are confirming the following to be true:_

- I have reviewed the [Contribution Guidelines](https://github.com/IndexCoop/index-app/blob/master/CONTRIBUTING.md).
- I have performed a self-review of my own code.
- I have updated my repository to match the master at `IndexCoop/index-app`.
- I have included test data or screenshots that prove my fix is effective or that my feature works.

### CODE REVIEW

File: src/app/leverage/provider.tsx [hunk 1]
```diff
@@ -152,7 +152,6 @@ export function LeverageProvider(props: { children: any }) {
   })
 
   const chainId = useMemo(() => {
-    updateQueryParams({ network: chainIdRaw })
     // To control the defaults better
     return queryNetwork ?? chainIdRaw ?? ARBITRUM.chainId
   }, [chainIdRaw, queryNetwork, updateQueryParams])
```
Comment (0xonramp) [inline src/app/leverage/provider.tsx:157]:
```suggestion
  }, [chainIdRaw, queryNetwork])
```

File: src/components/header/network-select.tsx [hunk 2]
```diff
@@ -27,6 +30,28 @@ export const NetworkSelect = () => {
     return AssetUtil.getNetworkImageById(currentNetwork?.imageId) ?? ''
   }, [networks, chainId])
 
+  useEffect(() => {
+    const unwatch = watchAccount(config, {
+      // Do not immediately override the chainId with the wallet one when the user arrives.
+      onChange({ chainId, isConnecting, isReconnecting }) {
+        console.log(isConnecting, isReconnecting)
```
Comment (0xonramp) [inline src/components/header/network-select.tsx:37]:
```suggestion
```

**Review by 0xonramp — COMMENTED** [review]
Two small nits, LGTM though

**Review by janndriessen — APPROVED** [review]
🚢

**Pull request merged**